src/compiler/crystal/semantic/exception.cr

```diff
@@ -171,6 +171,9 @@ module Crystal
     end
   end
 
+  class AbstractDefImplementationError < TypeException
```

Comment:
Why is this new type needed?

Reply:
It's not really needed, we can keep using TypeException if preferred. 

It's true that the others specialized exceptions appear in rescue or is_a? expressions.